service/controller/v2/resources/vaultcrt/current.go

```diff
@@ -38,10 +39,25 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		}
 	}
 
+	if key.IsInDeletionState(customObject) {
+		n := key.ClusterNamespace(customObject)
+		list, err := r.k8sClient.CoreV1().Pods(n).List(metav1.ListOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		if len(list.Items) != 0 {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cannot finish deletion of the secret due to existing pods")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
```

Comment:
So in case a cluster deletion happens, we delete all secrets holding certificates. We still need the certs for draining nodes on KVM though. So as long as pods are there we delay the deletion of the secrets here in order to still use them in the kvm-operator. The impact of this for AWS and Azure is zero. When listing on namespaces that do not exist we get an empty list and thus do nothing here. For KVM, as soon as the draining was done and the pods got removed we get an empty list here after the delete event got replayed and remove the secrets.

Reply:
I get the reasoning here, I would just really appreciate some comments in here. I doubt that I would remember this if I was reading this two weeks from now.

Reply:
Done.